ChopChop/Models/Ingredients/Quantity.swift

```diff
@@ -0,0 +1,258 @@
+/**
+ Represents the quantity of an ingredient.
+ 
+ Invariants:
+ - Quantities are non negative.
+ 
+ Mass type quantities are meaasured in kilograms.
+ Volume type quantities are measured in litres.
+ */
+enum Quantity {
+    case count(Double)
+    case mass(Double)
+    case volume(Double)
+
+    var type: QuantityType {
+        switch self {
+        case .count:
+            return .count
+        case .mass:
+            return .mass
+        case .volume:
+            return .volume
+        }
+    }
+
+    var value: Double {
+        get {
+            switch self {
+            case .count(let value):
+                return value
+            case .mass(let value):
+                return value
+            case .volume(let value):
+                return value
+            }
+        }
+        set {
+            switch self {
+            case .count:
+                self = .count(newValue)
+            case .mass:
+                self = .mass(newValue)
+            case .volume:
+                self = .volume(newValue)
+            }
+        }
+    }
+}
+
+// MARK: - Arithmetic operations
+extension Quantity {
+    /**
+     Returns the sum of two quantities if they are of the same type.
+     - Throws:
+        - `QuantityError.differentQuantityTypes`: if the types of the quantities do not match.
+        - `QuantityError.negativeQuantity`: if the result is negative.
+     */
+    static func + (left: Quantity, right: Quantity) throws -> Quantity {
+        switch (left, right) {
+        case let (.count(leftValue), .count(rightValue)):
+            let sum = leftValue + rightValue
+            guard sum >= 0 else {
+                throw QuantityError.negativeQuantity
+            }
```

Comment:
Is it better to enforce that quantities cannot be negative?

Reply:
Will be done when `Quantity` is redesigned to be a class